### PR TITLE
Arsenal - Fix incorrect muzzle velocity stat displayed on magazine

### PR DESCRIPTION
**Pull request by dedmen**
^ title

### CODE REVIEW

**Comment by dedmen** [comment]
@alganthe for review.
I would add a check whether advanced ballistics is enabled here, but that might not be set as the player expects in the Main Menu Arsenal. And also you don't need that value anyway if you have advanced ballistics disabled.

File: addons/ballistics/functions/fnc_statTextStatement_magazineMuzzleVelocity.sqf [hunk 1]
```diff
@@ -0,0 +1,44 @@
+#include "script_component.hpp"
+/*
+ * Author: Alganthe
+ * Text statement for the weapon muzzle velocity stat
```
Comment (jonpas) [inline addons/ballistics/functions/fnc_statTextStatement_magazineMuzzleVelocity.sqf]:
```suggestion
 * Text statement for the weapon muzzle velocity stat.
```

File: addons/ballistics/functions/fnc_statTextStatement_magazineMuzzleVelocity.sqf [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ *
+ * Arguments:
+ * 0: Not used
```
Comment (jonpas) [inline addons/ballistics/functions/fnc_statTextStatement_magazineMuzzleVelocity.sqf]:
```suggestion
 * 0: Type what it is here <TYPE> (unused)
```
Type what it is and add ` (unused)` at the end.

File: addons/ballistics/functions/fnc_statTextStatement_magazineMuzzleVelocity.sqf [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+ * 1: item config path (CONFIG)
```
Comment (jonpas) [inline addons/ballistics/functions/fnc_statTextStatement_magazineMuzzleVelocity.sqf]:
```suggestion
 * 1: Item config path <CONFIG>
```

File: addons/ballistics/functions/fnc_statTextStatement_magazineMuzzleVelocity.sqf [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+ *
+ * Return Value:
+ * String to display
```
Comment (jonpas) [inline addons/ballistics/functions/fnc_statTextStatement_magazineMuzzleVelocity.sqf]:
```suggestion
 * Display text <STRING>
```

**Comment by alganthe** [comment]
I didn't see the extra file because my browser wasn't centered....
make sure to remove it.

**Comment by dedmen** [comment]
I also didn't notice that tmp file when making the commit, wtf.

**Comment by jonpas** [comment]
> To jonpas suggestions, I just copied the weaponMuzzleVelocity file, should that also be updated?

Yeah, if you want.

File: addons/ballistics/functions/fnc_statTextStatement_magazineMuzzleVelocity.sqf [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    localize "str_empty";
+} else {
+    private _weaponCfg = configFile >> "CfgWeapons" >> _weapon;
+    private _ammoCfg = (configFile >> "CfgAmmo" >> (getText (configFile >> "CfgMagazines" >> _magazine >> "ammo")));
+    private _barrelLength = getNumber (_weaponCfg >> "ACE_barrelLength");
+    private _muzzleVelocityTable = getArray (_ammoCfg >> "ACE_muzzleVelocities");
+    private _barrelLengthTable = getArray (_ammoCfg >> "ACE_barrelLengths");
+
+    if (_barrelLength != 0 && {count _muzzleVelocityTable > 0} && {count _barrelLengthTable > 0}) then {
+        private _muzzleVelocity = [_barrelLength, _muzzleVelocityTable, _barrelLengthTable, 0] call EFUNC(advanced_ballistics,calculateBarrelLengthVelocityShift);
```
Comment (PabstMirror) [inline addons/ballistics/functions/fnc_statTextStatement_magazineMuzzleVelocity.sqf]:
```suggestion
        private _muzzleVelocity = if (["ace_advanced_ballistics"] call EFUNC(common,isModLoaded)) then {
            [_barrelLength, _muzzleVelocityTable, _barrelLengthTable, 0] call EFUNC(advanced_ballistics,calculateBarrelLengthVelocityShift);
        } else {
            getNumber (_config >> "initSpeed")
        };
```

Comment (PabstMirror) [reply addons/ballistics/functions/fnc_statTextStatement_magazineMuzzleVelocity.sqf]: no guarantee advanced_ballistics is loaded

**Comment by dedmen** [comment]
Remind me to reimplement https://github.com/acemod/ACE3/pull/6844#discussion_r262559079
in the weapon stat, if I forget tomorrow.

**Comment by dedmen** [comment]
Uh... @PabstMirror what?
https://github.com/acemod/ACE3/pull/6844#issuecomment-469751931

https://github.com/dedmen/ACE3/commit/9defee2e6eaf579e4cf2eb5a25a92516a9176e80

**Pull request merged**